apis/k8ssandra/v1alpha1/constants.go

```diff
@@ -64,8 +64,17 @@ const (
 
 	// Annotation to indicate the purpose of a given resource.
 	PurposeAnnotation = "k8ssandra.io/purpose"
+
+	// AutomatedUpdateAnnotation is an annotation that allows the Datacenters to be updated even if no changes were done to the K8ssandraCluster spec
+	AutomatedUpdateAnnotation = "k8ssandra.io/autoupdate-spec"
+
+	AllowUpdateAlways AllowUpdateType = "always"
+	AllowUpdateOnce   AllowUpdateType = "once"
 )
 
+// TODO Use the accepted values from cass-operator's api instead to prevent drift, once Kubernetes dependencies are updated in k8ssandra-operator
+type AllowUpdateType string
```

Comment:
That would blow up the PR, since updating cass-operator is not the issue, but the old dependencies in k8ssandra-operator.

So updating cass-operator requires updating Kubernetes dependencies, controller-runtime dependencies and those are not backwards compatible with the current code, so there would be more changes.

To give a scope of the required changes, here are the changes that cass-operator alone required: https://github.com/k8ssandra/cass-operator/commit/c273e274310b0b007178ab214b2e32774f3cf819